pynamodb/expressions/operand.py

```diff
@@ -248,6 +260,13 @@ def __init__(self, attribute_or_path: Union['Attribute', str, List[str]]) -> Non
             raise ValueError("path cannot be empty")
         super(Path, self).__init__(get_path_segments(path))
 
+    def _equals_to(self, other: Any) -> bool:
+        return (
+            super()._equals_to(other)
+            and self.attr_type == other.attr_type
+            and self.attribute is other.attribute
```

Comment:
`is` might be worthy of a comment for the `Attribute` case here

Reply:
to clarify that I'm doing `is` because it's a type?

Reply:
ah right, right